src/provider/model/Model.ts

```diff
@@ -91,6 +92,12 @@ export class Model {
     }
 
     public async expectWindow(identity: Identity): Promise<AppWindow> {
+        // Temporary work-around until .NET channel aliases are also applied to 'dispatch' calls, same as they are on channel connection events.
+        // Note that 'secondWindow' in .NET demo will not fully work whilst this workaround is in place.
+        if (identity.name && identity.name.split('-').length === 5) {
+            identity.name = 'mainwin';
+        }
```

Comment:
Will remove this before marking PR as ready, currently still doing some final testing - which requires this workaround.